application/libraries/Grocery_CRUD.php

```diff
@@ -2574,25 +2656,150 @@ protected function get_relation_n_n_input($field_info_type, $selected_values)
+
+							$extra_field->db_type = $extra_field->type;
+							$extra_field->db_max_length = $extra_field->max_length;
+
+							$extra_field_info = (object)array();
+							$extra_field_info->name		= $field_info_type->name.'Extras['.$id.']['.$extra_field->name.']';
+							$extra_field_info->crud_type 	= $this->get_type($extra_field);
+							$extra_field_info->type = $extra_field->type;
+							if($extra_field_info->crud_type== 'text'){
+								$extra_field_info->extras = 'text_editor';
+							}
+							elseif($extra_field_info->crud_type == 'set' || $extra_field_info->crud_type == 'enum'){
+								$extra_field_info->extras = $this->basic_model->enum_select($field_info_type->extras->relation_table, $extra_field->name);
+							}
+							else{
+								$extra_field_info->extras = null;
+							}
+							$extra_field_info->db_max_length = $extra_field->db_max_length;
+							$extra_field_info->required	= false;
+							$extra_field_info->display_as = ucfirst(str_replace("_"," ",$extra_field->name));
+							$input .= "<div>".$extra_field_info->display_as."</div><div>".$this->get_field_input($extra_field_info, $values->{$extra_field->name})->input."</div>";
+						}
+					}
+					$input .= "<p></p></div>";
+				}
+				$input .= "</div>";
+
+				$input .= "<div id='{$field_info_type->name}Template' style='display:none'>";
+
+				foreach ($extra_fields as $extra_field) {
+					if ( $this->unset_edit_fields !== null
+					&& is_array($this->unset_edit_fields)
+					&& in_array(('extra_field_'.$extra_field->name),$this->unset_edit_fields)
+					) {
+						continue;
+					}
+					if( $extra_field->name != $field_info_type->extras->primary_key_alias_to_this_table
+					&& $extra_field->name != $field_info_type->extras->primary_key_alias_to_selection_table){
+
+						$extra_field->db_type = $extra_field->type;
+						$extra_field->db_max_length = $extra_field->max_length;
+
+						$extra_field_info = (object)array();
+						$extra_field_info->name		= $field_info_type->name.'Extras[{primary_key_value}]['.$extra_field->name.']';
+						$extra_field_info->crud_type 	= $this->get_type($extra_field);
+						$extra_field_info->type = $extra_field->type;
+						if($extra_field_info->crud_type == 'text'){
+							$extra_field_info->extras = 'text_editor_to_be';
+						}
+						elseif($extra_field_info->crud_type == 'set' || $extra_field_info->crud_type == 'enum'){
+							$extra_field_info->extras = $this->basic_model->enum_select($field_info_type->extras->relation_table, $extra_field->name);
+						}
+						else{
+							$extra_field_info->extras = null;
+						}
+						$extra_field_info->db_max_length = $extra_field->db_max_length;
+						$extra_field_info->required	= false;
+						$extra_field_info->display_as = ucfirst(str_replace("_"," ",$extra_field->name));
+						$input .= "<div>".$extra_field_info->display_as."</div><div>".$this->get_field_input($extra_field_info)->input."</div>";
```

Comment:
//Add callback functionality 						
						if(!isset($this->callback_edit_field[$extra_field->name])){
							$input .= "<div>".$extra_field_info->display_as."</div><div>".$this->get_field_input($extra_field_info)->input."</div>";
						} else {
							$input .= call_user_func($this->callback_edit_field[$extra_field->name], null, $id, $extra_field_info);
						}